camp/camp-base/src/main/java/org/apache/brooklyn/camp/spi/pdp/AssemblyTemplateConstructor.java

```diff
@@ -40,15 +40,22 @@ public AssemblyTemplateConstructor(CampPlatform campPlatform) {
         this.transaction = this.campPlatform.transaction();
     }
     
-    /** records all the templates to the underlying platform */
+    /** records all the templates to the underlying platform 
+     * @deprecated since 0.10.0 use {@link #construct(boolean)} */
     public AssemblyTemplate commit() {
+        return construct(true);
+    }
+    
+    /** builds the template, optionally recording everything to the underlying platform */
+    public AssemblyTemplate construct(boolean save) {
         checkState();
         AssemblyTemplate at = builder.build();
-        transaction.add(at).commit();
+        if (!save) transaction.clear();
+        transaction.commit();
         transaction = null;
         return at;
     }
-
+    
```

Comment:
Can you just not call commit/construct (for the lack of rollback) and let the objects GC instead of doing the API changes in `camp-base`?